codice/ddf
# [2.13.x] DDF-4457 Encrypt file system storage

#### What does this PR do?
Updates the `FileSystemStorageProvider` to encrypt the content it writes to disk.

#### Who is reviewing it? 
@blen-desta 
@garrettfreibott 
@stustison 

#### Select relevant component teams: 
@codice/security 

#### How should this be tested?
1. Build DDF and install the standard profile.
1. Ingest a resource using Intrigue.
1. Query for that resource using Intrigue.
1. Verify that the result looks normal
    1. Check for a thumbnail (if applicable)
    1. Verify all of the `Actions` in the `Inspector` tab work, especially the `resource` action.
1. Verify that the content is encrypted on disk
   1. Navigate to `<DDF_HOME>/data/content/store/{first 3 characters of id}/{next three characters of id}/{id}`, and verify that you cannot open the resource.
1. Verify that the new keysets have been generated.
    1. Navigate to the keyset directory under `<DDF_HOME>/etc/keysets`
    1. Verify that there are 6 total entries: `encryption-persistence-manager.json`, `encryption-persistence-manager-streaming.json`, `encrypiton-service.json`, `encrypiton-service-streaming.json`, `file-system.json`, and `file-system-streaming.json`

#### What are the relevant tickets?
[DDF-4457](https://codice.atlassian.net/browse/DDF-4457)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
